p2p/net/swarm/dial_worker.go

```diff
@@ -326,20 +315,14 @@ loop:
 					continue loop
 				}
 
-				// request succeeded, respond to all pending requests
-				for _, reqno := range ad.requests {
-					pr, ok := w.pendingRequests[reqno]
-					if !ok {
-						// some other dial for this request succeeded before this one
-						continue
+				for pr := range w.pendingRequests {
```

Comment:
This loop will now be executed in random order. Is that ok?

Reply:
see https://github.com/libp2p/go-libp2p/pull/2416/files/06199d786a8323515bd18ba3e3ccb7b0db5704f8#r1258151128